import "libnutrition/proto/data/template.proto";
import "libnutrition/proto/service/error.proto";

message TemplateLoadRequest 
{
	optional bool omit = 1 [default = false];
	optional bool nameAndIdOnly = 2 [default = false];
	repeated int32 requestedIds = 3;
	repeated int32 requestedUserIds = 4;
}

message TemplateLoadResponse
{
	repeated TemplateData templates = 1;
	optional Error error = 100;
}

message TemplateStoreRequest
{
	repeated TemplateData templatess = 1;
}

message TemplateStoreResponse
{
	repeated int32 storedIds = 1;
	optional Error error = 100;
}

message TemplateDeleteRequest
{
	repeated int32 deleteIds = 1;
}

message TemplateDeleteResponse
{
	repeated int32 deletedIds = 1;
	optional Error error = 100;
}